### DISCUSSION TITLE
.net core 5 - getting 404 errors returned

### DISCUSSION THREAD

**Discussion by wallym** [dotnet/aspnetcore#25316] [General]
I've updated to Tuesday's release and the most recent VS 2019 16.8 preview 2.  My .net core app startups and does not generate any errors on startup using iis web dev.  The initial index page loads and everything is correct.  It is an asp .net core razor page named Index.cshtml.  It is within the /Pages directory of my project.  the images & css also load properly on that page.  The next page that I have is /Pages/PublicScores/Default.cshtml.  both locally and on azure, it returns a 404 page not found error.  I've tried a multitude of things to get this to work with no luck.  any suggestions are appreciated.  I have included my program and startup files to see if I am doing something wrong.  I assume that I am the problem, but I can't find it.  TIA.

    public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseStartup<Startup>();
                });
    }

    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {

            services.AddDbContext<ApplicationDbContext>(options =>
                options.UseSqlServer(
                    Configuration.GetConnectionString("DefaultConnection"),
                x => x.UseNetTopologySuite()));
            services.AddDefaultIdentity<AspNetUser>(options => options.SignIn.RequireConfirmedAccount = true)
                .AddEntityFrameworkStores<ApplicationDbContext>();
            services.AddRazorPages();            
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseDatabaseErrorPage();
            }
            else
            {
                app.UseExceptionHandler("/Error");
                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
                app.UseHsts();
            }

            app.UseHttpsRedirection();
            app.UseStaticFiles();

            app.UseRouting();

            app.UseAuthentication();
            app.UseAuthorization();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapRazorPages();
                endpoints.MapControllers();
            });
        }
    }

**Comment by Pilchie** [comment]
@javiercn - seems like a potential routing issue?

@wallym - if you turn logging up, do you get any useful information?

**Reply by javiercn** [reply]
@Pilchie not sure. 

I think something is likely missing on the app. This is very fundamental behavior that our tests would have caught.

@wallym can you file an issue instead and provide us with a minimal repro project (on a public github repo would be great) that illustrates the issue?

**Reply by wallym** [reply]
I've got some logging on.  My breakpoint in the first page of the app get's hit.  My /publicscores/default page never get's hit.  I'm logging to the console and don't see anything.

Just to be clear, this isn't an app that was working in the previous beta.  I am currently porting over some code and I just the other day got to the point where I could compile and test.

I'll try to come up with a minimal project.  You might need to give me until tomorrow for this.

**Reply by Pilchie** [reply]
If you change the log level in appsettings to informational, you might see additional logging from the various parts of the stack that give hints though.

**Reply by javiercn** [reply]
@wallym that's fine. There is no rush.

It's even better if you turn the log level all the way to Debug. Routing will indicate what things match and what things don't match. You can also do this https://github.com/dotnet/aspnetcore/blob/f157b78d93df0d65f5d6b4820f94a2fcd753b9e1/src/Mvc/test/WebSites/RoutingWebSite/Controllers/DebugController.cs

and use http://www.webgraphviz.com/ to visualize the route table

**Reply by wallym** [reply]
So, I created another web project.  I created a couple of cshtml pages, and they seem to be working.  I created a directory and a .cshtml file in it, and that displayed.  I'm going to keep playing with things and see what I find.

**Comment by wallym** [comment]
Also, I deployed my test app to another azure app service that I just created.  The app loaded properly.  I was also able to attach and debug.  So, it looks like things are working as is.  I've always felt that this problem was between my chair and my keyboard.  And I think it is.  I'm going to recreate it and try again.

**Comment by wallym** [comment]
I haven't quite solved this issue 100%, but I've gotten most of it.  Ultimately, I think the issue is one of my own creation, which I thought before.  I created a new project in VS16.8P2 this morning.  I'm able to see and do things.  i think that there is some layout issue on my end due to some nested layout pages in my original app, but that's on me.
